apis/python/src/tiledb/vector_search/index.py

```diff
@@ -59,6 +59,15 @@ def __init__(
         self.ctx = vspy.Ctx(config)
         self.group = tiledb.Group(self.uri, "r", ctx=tiledb.Ctx(config))
         self.storage_version = self.group.meta.get("storage_version", "0.1")
+        try:
+            self.distance_metric = vspy.DistanceMetric(
+                self.group.meta.get("distance_metric", vspy.DistanceMetric.L2)
+            )
+        except ValueError:
+            raise ValueError(
+                f"Invalid distance metric in metadata: {self.group.meta.get('distance_metric')}."
+            )
+
```

Comment:
Just checking, is the `ValueError` that will be thrown here from an incorrect metadata value being set and the `vspy.DistanceMetric()` constructor throwing?

Reply:
Yep, true

Reply:
Got it, and is there any way for that to happen today? And what error does it throw before you catch and re-throw? 

I ask because I'm not totally convinced we need this error handling as it seems like something that cannot happen, and if it does, will get automatically thrown... (and the other metadata reads don't do this)